addons/web_editor/static/lib/odoo-editor/src/utils/sanitize.js

```diff
@@ -167,6 +170,15 @@ class Sanitize {
                 node.setAttribute('contenteditable', 'false');
             }
         }
+        // update link URL if label is a new valid link
```

Comment:
I would not do it in a sanitize but only when the dom is being mutated. Maybe we can add a flag when calling sanitize from a mutation.